nixos/modules/services/web-apps/privatebin.nix

```diff
@@ -0,0 +1,274 @@
+            dsn = "pgsql:host=localhost;dbname=privatebin";
+            tbl = "privatebin_";
+            user = "privatebin";
+            pwd._env = "PRIVATEBIN_DB_PASS";
+          };
+        }
+      '';
+      description = lib.mdDoc ''
+        Privatebin configuration as outlined in
+        <https://github.com/PrivateBin/PrivateBin/wiki/Configuration>.
+
+        Available sections are `[main]`, `[expire]`, `[expire_options]`,
+        `[formatter_options]`, `[traffic]`, `[purge]`, `[model]`,
+        `[model_options]` and `[yourls]`.
+      '';
+      type = format.type;
+    };
+
+    environmentFiles = mkOption {
+      type = with types; listOf path;
+      description = lib.mdDoc "Optional files containing environment variables with secrets to be passed to the config";
+      default = [];
+    };
+
+    databaseSetup = {
+      enable = mkEnableOption (lib.mdDoc "Automatic database setup and configuration");
+      kind = mkOption {
+        type = types.enum [ "pgsql" "mysql" "sqlite" ];
+        description = lib.mdDoc "Type of database to automatically set up";
+      };
+    };
+
+    poolConfig = mkOption {
+      type = with types; attrsOf (oneOf [ str int bool ]);
+      default = {
+        "pm" = "ondemand";
+        "pm.max_children" = 10;
+        "pm.process_idle_timeout" = "30s";
+        "pm.max_requests" = 200;
+      };
+      description = lib.mdDoc ''
+        Options for the PrivateBin PHP pool. See the documentation on `php-fpm.conf`
+        for details on configuration directives.
+      '';
+    };
+
+    phpPackage = mkOption {
+      type = types.package;
+      relatedPackages = [ "php80" "php81" "php82" ];
+      default = pkgs.php81;
+      defaultText = "pkgs.php81";
+      description = lib.mdDoc ''
+        PHP package to use for this PrivateBin instance.
+      '';
+    };
+
+    phpOptions = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      description = lib.mdDoc ''
+        Options for PHP's php.ini file for this dokuwiki site.
+      '';
+      example = literalExpression ''
+        {
+          "opcache.interned_strings_buffer" = "8";
+          "opcache.max_accelerated_files" = "10000";
+          "opcache.memory_consumption" = "128";
+          "opcache.revalidate_freq" = "15";
+          "opcache.fast_shutdown" = "1";
+        }
+      '';
+    };
+
+    nginx = mkOption {
+      type = types.submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; });
+      default = { };
+      example = literalExpression ''
+        {
+          forceSSL = true;
+          enableACME = true;
+        }
+      '';
+      description = lib.mdDoc ''
+        Optional settings to pass to the nginx virtualHost.
+      '';
+    };
+  };
+
+  config = {
+    users.users.${user} = {
+      inherit group;
+
+      isSystemUser = true;
+      createHome = false;
+    };
+    users.users.${config.services.nginx.user}.extraGroups = [ group ];
```

Comment:
even if we disable nginx in our own config (i.e. by overriding the default set on line 249), this line still tries to add the nginx user to the privatebin group. Perhaps we could have this line conditional on nginx still being enabled?